pkgs/development/compilers/ghc/8.10.4.nix

```diff
@@ -105,6 +105,8 @@ let
     GhcRtsHcOpts += -fPIC
   '' + lib.optionalString targetPlatform.useAndroidPrebuilt ''
     EXTRA_CC_OPTS += -std=gnu99
+  '' + lib.optionalString targetPlatform.isWindows ''
+    SplitSections = NO
```

Comment:
Can you add a comment why this was done? It seems to me that `-split-sections` should work for windows, but is extremely slow when using `ld.bfd`.

Reply:
This was done because there was a "too many sections" error otherwise. Took the change from https://github.com/input-output-hk/haskell.nix/blob/531c9f4cac0e335db245849df3b5036973826d36/compiler/ghc/default.nix#L126-L128

Though after looking it into it a bit more, it seems this could be fixed with `"-Wa,-mbig-obj"`, but I can't figure out where to pass those flags.

Reply:
`-Wa,`? Or `-Wl,`?

Either way, `EXTRA_CC_OPTS` should work.

Reply:
```diff
-    SplitSections = NO
+    EXTRA_CC_OPTS += -Wa,-mbig-obj
```
yields same error

Reply:
Given the actual error message
```
"inplace/bin/ghc-stage1" -hisuf p_hi -osuf  p_o -hcsuf p_hc -static -prof -eventlog  -O -H64m -Wall     -this-unit-id array-0.5.4.0 -hide-all-packages -i -ilibraries/array/. -ilibraries/array/dist-install/build -Ilibraries/array/dist-install/build -ilibraries/array/dist-install/build/./autogen -Ilibraries/array/dist-install/build/./autogen -Ilibraries/array/.    -optP-include -optPlibraries/array/dist-install/build/./autogen/cabal_macros.h -package-id base-4.14.1.0 -Wall -XHaskell2010 -O2 -fPIC  -no-user-package-db -rtsopts  -Wno-deprecated-flags    -Wnoncanonical-monad-instances  -outputdir libraries/array/dist-install/build -split-sections  -c libraries/array/./Data/Array/MArray/Safe.hs -o libraries/array/dist-install/build/Data/Array/MArray/Safe.p_o 
/nix/store/aq9brwnas2npzk92jy4bdv9mqyc6dqzc-x86_64-w64-mingw32-binutils-2.35.1/bin/x86_64-w64-mingw32-ld: libraries/base/dist-install/build/HSbase-4.14.1.0.p_o: too many sections (55449)
/nix/store/aq9brwnas2npzk92jy4bdv9mqyc6dqzc-x86_64-w64-mingw32-binutils-2.35.1/bin/x86_64-w64-mingw32-ld: final link failed: file too big
make[1]: *** [libraries/base/ghc.mk:4: libraries/base/dist-install/build/HSbase-4.14.1.0.p_o] Error 1
make[1]: *** Waiting for unfinished jobs....
```
comes from `ld`, I tried
```diff
-    SplitSections = NO
+    EXTRA_LD_OPTS += -Wl,--oformat,pe-bigobj-x86-64
```
following https://stackoverflow.com/a/64367746/6276652. This fails fairly early on with 
```
/nix/store/hy3lz2vfv9qq2v5jz9nzlx6mmiaq79rj-binutils-2.35.1/bin/ld: target pe-bigobj-x86-64 not found
```
I tried forcing `"--enable-targets=all` in `binutils/default.nix`. This also fails early on with
```
"/nix/store/amw7sxk9qk4bsc8m1x71ifp3qcmw77j3-ghc-8.6.5-binary/bin/ghc" -o utils/deriveConstants/dist/build/tmp/deriveConstants -hisuf hi -osuf  o -hcsuf hc -static  -O -H64m -Wall   -package-db libraries/bootstrapping.conf  -hide-all-packages -i -iutils/deriveConstants/. -iutils/deriveConstants/dist/build -Iutils/deriveConstants/dist/build -iutils/deriveConstants/dist/build/deriveConstants/autogen -Iutils/deriveConstants/dist/build/deriveConstants/autogen     -optP-include -optPutils/deriveConstants/dist/build/deriveConstants/autogen/cabal_macros.h -package-id base-4.12.0.0 -package-id containers-0.6.0.1 -package-id filepath-1.4.2.1 -package-id process-1.6.5.0 -XHaskell2010  -no-user-package-db -rtsopts        -outputdir utils/deriveConstants/dist/build   -optl-Wl,--oformat,pe-bigobj-x86-64 -static  -O -H64m -Wall   -package-db libraries/bootstrapping.conf  -hide-all-packages -i -iutils/deriveConstants/. -iutils/deriveConstants/dist/build -Iutils/deriveConstants/dist/build -iutils/deriveConstants/dist/build/deriveConstants/autogen -Iutils/deriveConstants/dist/build/deriveConstants/autogen     -optP-include -optPutils/deriveConstants/dist/build/deriveConstants/autogen/cabal_macros.h -package-id base-4.12.0.0 -package-id containers-0.6.0.1 -package-id filepath-1.4.2.1 -package-id process-1.6.5.0 -XHaskell2010  -no-user-package-db -rtsopts        utils/deriveConstants/dist/build/Main.o    
"/nix/store/amw7sxk9qk4bsc8m1x71ifp3qcmw77j3-ghc-8.6.5-binary/bin/ghc" -hisuf hi -osuf  o -hcsuf hc -static  -O -H64m -Wall   -package-db libraries/bootstrapping.conf  -hide-all-packages -i -iutils/hsc2hs/. -iutils/hsc2hs/dist/build -Iutils/hsc2hs/dist/build -iutils/hsc2hs/dist/build/hsc2hs/autogen -Iutils/hsc2hs/dist/build/hsc2hs/autogen     -optP-include -optPutils/hsc2hs/dist/build/hsc2hs/autogen/cabal_macros.h -package-id base-4.12.0.0 -package-id containers-0.6.0.1 -package-id directory-1.3.3.0 -package-id filepath-1.4.2.1 -package-id process-1.6.5.0 -Wall -XHaskell2010  -no-user-package-db -rtsopts        -outputdir utils/hsc2hs/dist/build   -c utils/hsc2hs/./DirectCodegen.hs -o utils/hsc2hs/dist/build/DirectCodegen.o 
/nix/store/rmv3kghdvf3kzxdm85zywd8ibannlyly-binutils-2.35.1/bin/ld: skipping incompatible /nix/store/khj4v46ckd20q4gal00336ia6qhasbnb-glibc-2.33-47/lib/libm.so when searching for -lm
/nix/store/rmv3kghdvf3kzxdm85zywd8ibannlyly-binutils-2.35.1/bin/ld: skipping incompatible /nix/store/khj4v46ckd20q4gal00336ia6qhasbnb-glibc-2.33-47/lib/libm.so when searching for -lm
/nix/store/rmv3kghdvf3kzxdm85zywd8ibannlyly-binutils-2.35.1/bin/ld: cannot find -lm
/nix/store/rmv3kghdvf3kzxdm85zywd8ibannlyly-binutils-2.35.1/bin/ld: skipping incompatible /nix/store/khj4v46ckd20q4gal00336ia6qhasbnb-glibc-2.33-47/lib/libm.so when searching for -lm
/nix/store/rmv3kghdvf3kzxdm85zywd8ibannlyly-binutils-2.35.1/bin/ld: skipping incompatible /nix/store/khj4v46ckd20q4gal00336ia6qhasbnb-glibc-2.33-47/lib/libm.so when searching for -lm
/nix/store/rmv3kghdvf3kzxdm85zywd8ibannlyly-binutils-2.35.1/bin/ld: /nix/store/khj4v46ckd20q4gal00336ia6qhasbnb-glibc-2.33-47/lib/librt.so: error adding symbols: file in wrong format
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
``` 

Not sure how to proceed (or even if I'm going in the right direction). 
I'm tempted to leave `SplitSections` disabled with a comment.